rpcs3/Emu/VFS.cpp

```diff
@@ -701,18 +705,72 @@ std::string vfs::unescape(std::string_view name)
 	return result;
 }
 
-bool vfs::host::rename(const std::string& from, const std::string& to, bool overwrite)
+bool vfs::host::rename(const std::string& from, const std::string& to, const lv2_fs_mount_point* mp, bool overwrite)
 {
-	while (!fs::rename(from, to, overwrite))
+	// Try the easy way
+	for (int i = 0; i < 5; i++, std::this_thread::yield())
 	{
-		// Try to ignore access error in order to prevent spurious failure
-		if (Emu.IsStopped() || fs::g_tls_error != fs::error::acces)
+		if (fs::rename(from, to, overwrite))
+		{
+			return true;
+		}
+
+		if (fs::g_tls_error != fs::error::acces)
 		{
 			return false;
 		}
 	}
 
-	return true;
+#ifdef _WIN32
+	// Hard way: lock mount point, close file descriptors, retry 
+	const auto from0 = from.substr(0, from.find_last_not_of("/\\") + 1) + '/';
+
+	std::lock_guard lock(mp->mutex);
+
+	idm::select<lv2_fs_object, lv2_file>([&](u32 id, lv2_file& file)
+	{
+		if (file.mp == mp && vfs::get(file.name.data()).starts_with(from0))
+		{
+			file.file.close(); // Actually close it!
```

Comment:
Oh rip. Need to use seek to save/restore current position...

Reply:
Thanks, fixed.